Automattic/wp-calypso
# Recurring Payments: Split out edit modal.

The "Add new plan" and "Edit" actions for payments plans (at `/earn/payments-plans/:site`) are coupled together through state and sharing a dialog in `MembershipsProductsSection`. This causes the "Add new plan" dialog to be briefly visible when dismissing the edit version of the dialog. By splitting out the dialog and handling dialog openings individually, we should end up with simpler payments plans components, and better rendering of each modal dialog.

<img width="1470" alt="Screen Shot 2020-05-21 at 4 51 04 PM" src="https://user-images.githubusercontent.com/349751/82617150-0d56c480-9b84-11ea-9181-1fea5f2033e5.png">



#### Testing instructions

* Load this branch with a site that has a Personal plan or higher.
* Go to `/earn/payments-plans/:site`.
* Add a new plan, and verify that everything works as expected.
* Load an existing plan (via the ellipsis menu Edit button), and verify initial data loads correctly, and saves correctly.
* Verify there is no flash of different modals being visible at the same time.
